clients/src/main/java/org/apache/kafka/clients/producer/internals/Sender.java

```diff
@@ -285,7 +285,7 @@ public void forceClose() {
 
     private ClientResponse sendAndAwaitInitPidRequest(Node node) throws IOException {
         String nodeId = node.idString();
-        InitPidRequest.Builder builder = new InitPidRequest.Builder(null);
+        InitPidRequest.Builder builder = new InitPidRequest.Builder(null, Integer.MAX_VALUE);
```

Comment:
@apurvam there already is an overload, but it also sets the timeout to `Integer.MAX_VALUE`